### PR TITLE
DruidHook add SQL-based task support

### PR DESCRIPTION
**Pull request by victorymakes**
### What
DruidHook adds SQL-based task support.

### Why
The current DruidHook does not support submitting the SQL-based task. The `submit_indexing_job` uses the same endpoint for submitting and checking the job status. However, the SQL-based ingestion job uses different endpoints for submitting and checking the job status.
- https://docs.imply.io/latest/druid/api-reference/sql-ingestion-api/#submit-a-query

### CODE REVIEW

**Review by Lee-W — COMMENTED** [review]
As they're using different endpoints, I wonder whether we should create a new hook and operator instead of overriding the parameter.

File: airflow/providers/apache/druid/hooks/druid.py
Comment (pateash) [inline airflow/providers/apache/druid/hooks/druid.py]:
```suggestion
```

Comment (pateash) [reply airflow/providers/apache/druid/hooks/druid.py]: Please fix

**Comment by vioao** [comment]
> As they're using different endpoints, I wonder whether we should create a new hook and operator instead of overriding the parameter.

@Lee-W 
I am not sure about this.  But I think both of them doing the same things, they could be put on the same hook. 
Although the SQL-based ingestion task does not follow the original URL format.
It uses the `/druid/v2/sql/task` endpoint to submit the task, but it uses the `/druid/indexer/v1/task/<taskId>/status` endpoint to check the task status.

**Comment by Lee-W** [comment]
@vioao Or maybe we can make the new hook a subclass of the original one? WDYT?

**Comment by vioao** [comment]
@Lee-W I thought about it. 
But it also needs to change the current `DruidOperator` or add a new one. At the same time, the current `DruidHook` also needs to be reformed(eg. extract the status-checking method for common usage). 
I want to make the changes to be as small as possible.

**Comment by vioao** [comment]
@Lee-W I refactor the implementation. Instead of using the endpoint hard code, I add a new parameter `IngestionType` with a default value.

**Review by Lee-W — APPROVED** [review]
Great refactor. I thought I approved it yesterday

**Comment by vioao** [comment]
Thanks. But it seems it is still blocked. What more needs to be done to merge？

**Comment by Lee-W** [comment]
might need some help from committers. @pankajastro @phanikumv Would you be able to take a look at this one? Thanks!

**Comment by phanikumv** [comment]
> might need some help from committers. @pankajastro @phanikumv Would you be able to take a look at this one? Thanks!

just approved the workflow to run

**Comment by Lee-W** [comment]
@vioao it seems the test fails. Could you please take a look? Thanks!

**Comment by vioao** [comment]
@Lee-W Sorry about that. I fixed the ut error.  @pankajastro @phanikumv Could you have a look too? Thanks

**Comment by vioao** [comment]
@pankajastro @phanikumv Could you have a look to unblock the workflow?

**Review by pankajastro — APPROVED** [review]
LGTM

**Review by eladkal — COMMENTED** [review]
Can you please update the [docs](https://github.com/apache/airflow/blob/2b92c3c74d3259ebac714f157c525836f0af50f0/docs/apache-airflow-providers-apache-druid/operators.rst#L19) with some explanation about this new functionality?

**Comment by vioao** [comment]
@eladkal Add the doc and the parameter desc.

**Review by phanikumv — COMMENTED** [review]
can you also add an example for using ingestion type in `/tests/system/providers/apache/druid/example_druid_dag.py`

**Comment by vioao** [comment]
@phanikumv It seems the example only includes the necessary parameters, I do know whether should I add the `ingestion_type`  parameter as it has the default value. It looks a little redundant.

**Comment by vioao** [comment]
@eladkal @phanikumv Could you review and make the process work?

**Review by potiuk — APPROVED** [review]

**Pull request merged**
